src/jrd/replication/Publisher.cpp

```diff
@@ -102,18 +98,33 @@ namespace
 			const auto manager = dbb->replManager();
 			const auto& guid = dbb->dbb_guid;
 			const auto& userName = attachment->att_user->getUserName();
+			const auto config = manager->getConfig();
 
-			attachment->att_replicator = (IReplicatedSession*) FB_NEW_POOL(pool)
-				Replicator(pool, manager, guid, userName, cleanupTransactions);
-		}
+			if (config->pluginName == "builtin")
+			{
+				attachment->att_replicator = FB_NEW
+					Replicator(pool, manager, guid, userName, cleanupTransactions);
+			}
+			else
+			{
+				GetPlugins<IReplicatedSession> plugins(IPluginManager::TYPE_REPLICATOR, config->pluginName.c_str());
+				if (!plugins.hasData())
+				{
+					string msg;
+					msg.printf("Replication plugin %s not found\n\t%s", config->pluginName.c_str(), LOG_ERROR_MSG);
+					logOriginMessage(dbb->dbb_filename, msg, ERROR_MSG);
+					return nullptr;
+				}
+				attachment->att_replicator = plugins.plugin();
+			}
 
-		fb_assert(attachment->att_replicator);
+			if (attachment->att_replicator.hasData())
+				attachment->att_replicator->setAttachment(attachment->getInterface());
+		}
```

Comment:
Plugin factory returning nullptr.

Reply:
Given the condition if (!plugins.hasData()) above, the factory cannot return nullptr, AFAIU.

Reply:
Yes, you are right.
But still internal replication plugin creation can be skipped because of missing configuration settings ending up in att_replicator containing nullptr so the check here is needed bu assert below is better to be removed.